auto_nag/scripts/variant_expiration.py

```diff
@@ -0,0 +1,301 @@
+        """
+        super().__init__()
+
+        self.variants = self.get_variants()
+        today = lmdutils.get_date_ymd("today")
+        self.open_bug_date = lmdutils.get_date_ymd(today + timedelta(open_bug_days))
+        self.reminder_date = lmdutils.get_date_ymd(today + timedelta(reminder_days))
+        self.cc_on_bugs = list(cc_on_bugs)
+
+    def description(self) -> str:
+        return "Variants that need to be updated or dropped"
+
+    def has_default_products(self) -> bool:
+        return False
+
+    def has_product_component(self):
+        return True
+
+    def columns(self):
+        return ["id", "product", "component", "variant_id", "expiration", "action"]
+
+    def sort_columns(self):
+        # sort by expiration date
+        return lambda x: [4]
+
+    def escalate(self, person, priority, **kwargs):
+        # Escalate based on the number of days since the variant expiration date
+        days = (kwargs["expiration_date"] - self.nag_date).days
+        return self.escalation.get_supervisor(priority, days, person, **kwargs)
+
+    def get_variants(self) -> dict:
+        """Get the variants from the variants.yml file"""
+
+        resp = requests.get(VARIANTS_HG_URL, timeout=20)
+        resp.raise_for_status()
+
+        variants = yaml.safe_load(resp.text)
+        for variant in variants.values():
+            expiration = variant["expiration"]
+
+            if expiration == "never":
+                expiration = datetime.max
+
+            variant["expiration"] = lmdutils.get_date_ymd(expiration)
+
+        return variants
+
+    def get_bugs(self, date="today", bug_ids=[], chunk_size=None) -> dict:
+        bugs = super().get_bugs(date, bug_ids, chunk_size)
+
+        # Create bugs for variants that will be expired soon
+        for variant_id, variant_info in self.variants.items():
+            if (
+                variant_info.get("bug_id")
+                or variant_info["expiration"] >= self.open_bug_date
+            ):
+                continue
+
+            component = ComponentName.from_str(variant_info["component"])
+            expiration = variant_info["expiration"].strftime("%Y-%m-%d")
+            new_bug = {
+                "summary": f"The variant `{variant_id}` expiration is on {expiration}",
+                "product": component.product,
+                "component": component.name,
+                "status_whiteboard": "[variant-expiration]",
+                "type": "task",
+                "see_also": self.get_related_bug_ids(variant_id),
+                "cc": self.cc_on_bugs,
+                "description": BUG_DESCRIPTION,
+            }
+
+            if self.dryrun or self.test_mode:
+                bug = {"id": f"to be created for {variant_id}"}
+                logger.info(
+                    "A new bug for `%s` will be created with:\n%s", variant_id, new_bug
+                )
+            else:
+                bug = utils.create_bug(new_bug)
```

Comment:
should we have error handling here on create_bug?  specifically I see a component being renamed or retired and not updating variants.yml, then the bug create would fail.  I am not sure if create_bug does a system.exit() or something like that.